src/test/java/uk/gov/companieshouse/api/testdata/service/impl/AcspProfileServiceImplTest.java

```diff
@@ -83,6 +92,142 @@ void createAcspProfileWithDefaultValues() throws DataException {
+        verify(repository).save(captor.capture());
+
+        AcspProfile captured = captor.getValue();
+        assertEquals("randomId", captured.getId());
+        assertEquals("randomId", captured.getAcspNumber());
+        assertEquals(spec.getStatus(), captured.getStatus());
+        assertEquals(spec.getType(), captured.getType());
+        assertEquals("Test Data Generator randomId Company Ltd", captured.getName());
+        assertEquals("/authorised-corporate-service-providers/randomId", captured.getLinksSelf());
+        assertEquals(0L, captured.getVersion());
+
+        assertNotNull(captured.getAmlDetails());
+        assertEquals(3, captured.getAmlDetails().size());
+        assertEquals("Supervisory Body 1",
+                captured.getAmlDetails().get(0).getSupervisoryBody());
+        assertEquals("Membership Details 1",
+                captured.getAmlDetails().get(0).getMembershipDetails());
+        assertEquals("Supervisory Body 2",
+                captured.getAmlDetails().get(1).getSupervisoryBody());
+        assertEquals("Membership Details 2",
+                captured.getAmlDetails().get(1).getMembershipDetails());
+        assertEquals("Supervisory Body 3",
+                captured.getAmlDetails().get(2).getSupervisoryBody());
+        assertEquals("Membership Details 3",
+                captured.getAmlDetails().get(2).getMembershipDetails());
+    }
+
+    private AcspProfile getAcspProfile(AcspProfileSpec spec) {
+        AcspProfile savedProfile = createSavedProfile();
+
+        List<AmlDetails> amlDetailsList = new ArrayList<>();
+        for (AmlSpec amlSpec : spec.getAmlDetails()) {
+            AmlDetails amlDetails = new AmlDetails();
+            amlDetails.setSupervisoryBody(amlSpec.getSupervisoryBody());
+            amlDetails.setMembershipDetails(amlSpec.getMembershipDetails());
+            amlDetailsList.add(amlDetails);
+        }
+        savedProfile.setAmlDetails(amlDetailsList);
+        return savedProfile;
+    }
+
+    private AcspProfileSpec getAcspProfileSpec() {
+        AcspProfileSpec spec = new AcspProfileSpec();
+        spec.setStatus("active");
+        spec.setType("ltd");
+
+        // Creating multiple AML details entries
+        AmlSpec amlSpec1 = new AmlSpec();
+        amlSpec1.setSupervisoryBody("Supervisory Body 1");
+        amlSpec1.setMembershipDetails("Membership Details 1");
+
+        AmlSpec amlSpec2 = new AmlSpec();
+        amlSpec2.setSupervisoryBody("Supervisory Body 2");
+        amlSpec2.setMembershipDetails("Membership Details 2");
+
+        AmlSpec amlSpec3 = new AmlSpec();
+        amlSpec3.setSupervisoryBody("Supervisory Body 3");
+        amlSpec3.setMembershipDetails("Membership Details 3");
+
+        spec.setAmlDetails(List.of(amlSpec1, amlSpec2, amlSpec3));
+        return spec;
+    }
+
+    @Test
+    void createAcspProfileWithNullAmlDetails() throws DataException {
```

Comment:
What's the difference with the `createAcspProfile` test? I think they are doing the same and could just add the extra assertions to the existing one?